### ISSUE TITLE
Problem port forwards

### ISSUE THREAD

**Issue by michelebossa** [mammuth/ha-fritzbox-tools#24]
Hi,
I m trying to enable the port forwards switch but i receive an error into HA log "[custom_components.fritzbox_tools.switch] Port switches could not be enabled. Check if your fritzbox is able to do port forwardings!"

Can you help me?

Ha configuration:
![image](https://user-images.githubusercontent.com/38975058/67076058-b53e8680-f18c-11e9-9baa-dd4ec192ea64.png)

Friz config on HA server
![image](https://user-images.githubusercontent.com/38975058/67076139-dacb9000-f18c-11e9-980e-a5d5d75965c0.png)

Thanks

**Comment by michelebossa** [comment]
I have a dedicated IP and i have tried the https://www.wieistmeineip.de/ipv6-test/ and give me these results.
![image](https://user-images.githubusercontent.com/38975058/67078118-f33da980-f190-11e9-8b1f-9dad092cb5a8.png)

Do you think i can do something?
Thanks.

**Comment by AaronDavidSchneider** [comment]
that's strange and probably a bug of our code.
Which fritzbox do you use?

**Comment by michelebossa** [comment]
FRITZ!Box 4040

**Comment by AaronDavidSchneider** [comment]
Could you please change line 32 and 33 of `custom_component/switch.py` from
```
except:
            _LOGGER.error('Port switches could not be enabled. Check if your fritzbox is able to do port forwardings!')
```
to:
```
except Exception as e:
            _LOGGER.error('Port switches could not be enabled. Check if your fritzbox is able to do port forwardings!', e)
```
then restart homeassistant and post the new error.

**Comment by michelebossa** [comment]
I hope can help 👍 

2019-10-18 10:45:27 ERROR (MainThread) [homeassistant.components.switch] Error while setting up platform fritzbox_tools
Traceback (most recent call last):
  File "/config/custom_components/fritzbox_tools/switch.py", line 24, in setup_platform
    port_forwards_count: int = fritzbox_tools.connection.call_action('WANIPConnection:1', 'GetPortMappingNumberOfEntries')["NewPortMappingNumberOfEntries"]
KeyError: 'NewPortMappingNumberOfEntries'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 150, in _async_setup_platform
    await asyncio.wait_for(asyncio.shield(task), SLOW_SETUP_MAX_WAIT)
  File "/usr/local/lib/python3.7/asyncio/tasks.py", line 442, in wait_for
    return fut.result()
  File "/usr/local/lib/python3.7/concurrent/futures/thread.py", line 57, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/config/custom_components/fritzbox_tools/switch.py", line 33, in setup_platform
    _LOGGER.error('Port switches could not be enabled. Check if your fritzbox is able to do port forwardings!', e)
  File "/usr/local/lib/python3.7/logging/__init__.py", line 1407, in error
    self._log(ERROR, msg, args, **kwargs)
  File "/usr/local/lib/python3.7/logging/__init__.py", line 1514, in _log
    self.handle(record)
  File "/usr/local/lib/python3.7/logging/__init__.py", line 1524, in handle
    self.callHandlers(record)
  File "/usr/local/lib/python3.7/logging/__init__.py", line 1586, in callHandlers
    hdlr.handle(record)
  File "/usr/local/lib/python3.7/logging/__init__.py", line 894, in handle
    self.emit(record)
  File "/usr/src/homeassistant/homeassistant/components/system_log/__init__.py", line 176, in emit
    record, stack, _figure_out_source(record, stack, self.hass)
  File "/usr/src/homeassistant/homeassistant/components/system_log/__init__.py", line 102, in __init__
    self.message = record.getMessage()
  File "/usr/local/lib/python3.7/logging/__init__.py", line 369, in getMessage
    msg = msg % self.args
TypeError: not all arguments converted during string formatting

**Comment by mammuth** [comment]
Thanks for the great debugging @michelebossa and @AaronDavidSchneider! :kissing_heart:

**Comment by AaronDavidSchneider** [comment]
@michelebossa: I created a pull request to fix the issue. Could you please test the new code?
If you need help: go to https://github.com/AaronDavidSchneider/ha-fritzbox-tools/tree/fix_ports and  replace your `switch.py` with the one there.

**Comment by michelebossa** [comment]
It's working!!! Thanks a lot :)
 
![image](https://user-images.githubusercontent.com/38975058/67114175-9ddaba00-f1db-11e9-870f-c162e6c354f0.png)

**Comment by mammuth** [comment]
Awesome, thanks for the debugging @michelebossa and @AaronDavidSchneider! 😍

**Comment by mammuth** [comment]
Closed by #25

**Comment by mammuth** [comment]
@michelebossa I published version 1.2.4 in case you want to update it via HACS.

**Comment by michelebossa** [comment]
Thanks
